File: Data Structures & Algorithms/binary-search/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length - 1;
        while (l <= r){
            int idx = (l + r) / 2;
            if (nums[idx] == target){
                return idx;
            } else if (nums[idx] < target){
                l = idx + 1;
            } else {
                r = idx - 1;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        '''
        we can add it to a set
        '''
        numbers = set(nums)

        longest = 0

        for num in numbers:
            #instead of checking for longest streak every time.
            # we check for when the streak starts
            if (num - 1) not in numbers:
                length = 1
                while num + length in numbers:
                    length += 1
                longest = max(length, longest)
        return longest
            
